cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -300,12 +299,17 @@ char* gdv_fn_dec_to_string(int64_t context, int64_t x_high, uint64_t x_low,
     return val;                                                                      \
   }
 
-CAST_NUMERIC_FROM_STRING(int32_t, arrow::Int32Type, INT)
-CAST_NUMERIC_FROM_STRING(int64_t, arrow::Int64Type, BIGINT)
-CAST_NUMERIC_FROM_STRING(float, arrow::FloatType, FLOAT4)
-CAST_NUMERIC_FROM_STRING(double, arrow::DoubleType, FLOAT8)
+CAST_NUMERIC_STRING(int32_t, arrow::Int32Type, INT, utf8)
```

Comment:
Can you change the macros to CAST_NUMERIC_FROM_VARLEN_TYPES, another two macros CAST_NUMERIC_FROM_STRING and CAST_NUMERIC_FROM_BINARY which calls the above macro with utf8 or binary. Call these two macros with numberic types

Reply:
done